src/utils/body.ts

```diff
@@ -50,33 +84,93 @@ async function parseFormData<T extends BodyData = BodyData>(
   return {} as T
 }
 
+/**
+ * Converts form data to body data based on the provided options.
+ *
+ * @template T - The type of the parsed body data.
+ * @param {FormData} formData - The form data to convert.
+ * @param {ParseBodyOptions} options - Options for parsing the form data.
+ * @returns {T} The converted body data.
+ */
 function convertFormDataToBodyData<T extends BodyData = BodyData>(
   formData: FormData,
   options: ParseBodyOptions
 ): T {
-  const form: BodyData = {}
+  const form: BodyData = Object.create(null)
 
   formData.forEach((value, key) => {
     const shouldParseAllValues = options.all || key.endsWith('[]')
 
-    if (!shouldParseAllValues) {
-      form[key] = value
-    } else {
+    if (shouldParseAllValues) {
       handleParsingAllValues(form, key, value)
+    } else {
+      form[key] = value
     }
   })
 
+  if (options.dot) {
+    const nestedForm: BodyData = Object.create(null)
+
+    Object.entries(form).forEach(([key, value]) => {
+      const shouldParseDotValues = key.includes('.')
+
+      if (shouldParseDotValues) {
+        handleParsingNestedValues(nestedForm, key, value)
+      } else {
+        nestedForm[key] = value
+      }
+    })
+
+    return nestedForm as T
+  }
+
   return form as T
 }
 
+/**
+ * Handles parsing all values for a given key, supporting multiple values as arrays.
+ *
+ * @param {BodyData} form - The form data object.
+ * @param {string} key - The key to parse.
+ * @param {FormDataEntryValue} value - The value to assign.
+ */
 const handleParsingAllValues = (form: BodyData, key: string, value: FormDataEntryValue): void => {
-  const formKey = form[key]
-
-  if (formKey && Array.isArray(formKey)) {
-    ;(form[key] as (string | File)[]).push(value)
-  } else if (formKey) {
-    form[key] = [formKey, value]
+  if (form[key] !== undefined) {
+    if (Array.isArray(form[key])) {
+      ;(form[key] as (string | File)[]).push(value)
+    } else {
+      form[key] = [form[key] as string | File, value]
+    }
   } else {
     form[key] = value
   }
 }
+
+/**
+ * Handles parsing nested values using dot notation keys.
+ *
+ * @param {BodyData} form - The form data object.
+ * @param {string} key - The dot notation key.
+ * @param {FormDataEntryValue} value - The value to assign.
+ * @param {boolean} parseAllValues - Whether to parse all values as arrays.
```

Comment:
Sorry for the nit-picking but now you can remove this line from the TSDocs and then do the denofy thing

Reply:
Thanks, it my mistake actually, focusing on multiple work 😄